### PR TITLE
[PREVIEW] Scala removal from server/hotrod module

### PR DESCRIPTION
**Pull request by wburns**
This doesn't have a fully converted test directory yet, but it should all compile.

### CODE REVIEW

**Comment by wburns** [comment]
A bunch of tests will fail until I fix `HotRodClient.decode` method.

**Comment by wburns** [comment]
This should be good to go.  I can do the tests in a separate PR.

**Comment by gustavonalle** [comment]
Looking

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+   Response notExistResp() {
+      return decoder.createNotExistResponse(header);
+   }
+
+   Response createGetResponse(CacheEntry<byte[], byte[]> entry) {
+      return decoder.createGetResponse(header, entry);
+   }
+
+//   Object createMultiGetResponse(pairs: Map[Bytes, CacheEntry[Bytes, Bytes]]): AnyRef = null
+
```
Comment (gustavonalle) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java]: Remove?

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/ServerAddress.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      if (port != that.port) return false;
+      return host.equals(that.host);
+
+   }
+
+//   // IMPORTANT NOTE: Hot Rod protocol agrees to this calculation for a node
+//   // address hash code calculation, so any changes to the implementation
+//   // require modification of the protocol.
+//   override def hashCode() = Arrays.hashCode(
+//      "%s:%d".format(host, port).getBytes(UTF8))
```
Comment (gustavonalle) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/ServerAddress.java]: Remove?

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/test/AbstractTestTopologyAwareResponse.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.infinispan.server.hotrod.ServerAddress;
+
+import java.util.Collection;
+
+/**
+ * @author wburns
+ * @since 9.0
+ */
+public abstract class AbstractTestTopologyAwareResponse {public final int topologyId;
```
Comment (gustavonalle) [inline server/hotrod/src/test/java/org/infinispan/server/hotrod/test/AbstractTestTopologyAwareResponse.java]: style

Comment (gustavonalle) [inline server/hotrod/src/test/java/org/infinispan/server/hotrod/test/AbstractTestTopologyAwareResponse.java]: Actually @wburns could you run a reformat in all files?

**Comment by gustavonalle** [comment]
I am getting a deadlock when running the ```infinispan-server-core```:

Here's the relevant stack trace:

```
"testng-AbstractProtocolServerTest" #14 prio=5 os_prio=0 tid=0x00007f2a1c70e000 nid=0x6f95 in Object.wait() [0x00007f29fdaf7000]
   java.lang.Thread.State: RUNNABLE
        at io.netty.util.internal.SystemPropertyUtil.<clinit>(SystemPropertyUtil.java:38)
        at io.netty.channel.epoll.Native.<clinit>(Native.java:56)
        at io.netty.channel.epoll.Epoll.<clinit>(Epoll.java:33)
        at org.infinispan.server.core.AbstractProtocolServer.<clinit>(AbstractProtocolServer.java:47)


"TestNG" #12 prio=5 os_prio=0 tid=0x00007f2a1c6f2800 nid=0x6f92 in Object.wait() [0x00007f29fdcf9000]
   java.lang.Thread.State: RUNNABLE
        at io.netty.util.internal.ThreadLocalRandom.<clinit>(ThreadLocalRandom.java:67)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at io.netty.util.internal.logging.InternalLoggerFactory.<clinit>(InternalLoggerFactory.java:44)
        at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:67)
        at io.netty.buffer.UnpooledByteBufAllocator.<clinit>(UnpooledByteBufAllocator.java:30)
        at io.netty.buffer.Unpooled.<clinit>(Unpooled.java:73)
        at org.infinispan.server.core.VariableLengthTest.writeReadInt(VariableLengthTest.java:116)
        at org.infinispan.server.core.VariableLengthTest.test2pow14(VariableLengthTest.java:37)
```

the RUNNABLE above is misleading because the threads are actually blocked in the static initializers of the classes inside Netty

**Comment by gustavonalle** [comment]
@wburns looks good overall, just need to reformat code and wrt the deadlock, not sure if it as master problem as well, but apparently CI has been running fine lately

**Comment by gustavonalle** [comment]
@wburns trying hard to reproduce the issue again but can't. Looks like I hit a jackpot the 1st time I was running the full suite :smile: 
Does it happen to you?

**Comment by gustavonalle** [comment]
@wburns report this on https://github.com/netty/netty/issues/5720 it will be fixed when netty 4.1.5.Final is released.

In the mean time I think a 
```
<infinispan.test.parallel.threads>1</infinispan.test.parallel.threads>
```
would prevent the issue.

**Comment by anistor** [comment]
@wburns Hopefully once scala is completely removed we can get rid of all those manually produced service provider files located under resources/META-INF/services and generate them automatically based on the ```@MetaInfServices``` annotation and the metainf-services maven plugin. This is not currently incompatible with scala.

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java (hunk 1 above)
Comment (wburns) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java]: Hrmm let me look.

Comment (wburns) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java]: Yeah it was never used before either :P.  Just wanted to confirm I didn't forget this, heh.

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/ServerAddress.java (hunk 2 above)
Comment (wburns) [inline server/hotrod/src/main/java/org/infinispan/server/hotrod/ServerAddress.java]: This was actually something I need to look into, thanks for reminding me.  This code was here before like this but commented out.  It talks about the hashCode requiring a specific implementation, which the old code didn't do.  I wanted to make sure that was alright before I removed it.

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/test/AbstractTestTopologyAwareResponse.java (hunk 3 above)
Comment (wburns) [inline server/hotrod/src/test/java/org/infinispan/server/hotrod/test/AbstractTestTopologyAwareResponse.java]: Sure I can try updating the whole module :)

**Comment by wburns** [comment]
@gustavonalle Core should be a small enough package to disable multi threading, but I worry this could happen in other modules as well which we can't afford to disable multi threaded tests :(  Did you want me to still disable it for core though?  Thinking it will be a very small bandaid.

@anistor Sounds good to me, I am not as familiar with how this is changed, maybe I can just make another JIRA.

**Comment by wburns** [comment]
I have removed the commented out method and fixed all the code styles (separate commit - squash when pushing) and it should be ready.

**Comment by gustavonalle** [comment]
@wburns this specific deadlock is a test issue only when netty is involved , during normal server operation it will problably not happen (this was fixed on https://github.com/infinispan/infinispan/pull/4485), so a threading of "1" would not affect test times and prevent the issue for the moment

**Comment by anistor** [comment]
@wburns If you create the jira I promise to fix that :)

**Comment by wburns** [comment]
All updated.

**Comment by gustavonalle** [comment]
Looks good, pulling

**Comment by gustavonalle** [comment]
Merged, thanks a lot @wburns !

**Pull request closed without merging**